android-kotlin/common/src/main/java/com/google/android/googlevideodiscovery/common/models/VideoEntity.kt

```diff
@@ -56,24 +49,71 @@ data class MovieEntity(
 data class TvEpisodeEntity(
     override val id: String,
     override val name: String,
-    override val duration: Duration,
-    override val playbackUris: PlatformSpecificUris,
-    override val images: List<Image>,
-    override val releaseYear: Int,
-    override val genre: String,
+    val images: List<Image>,
+    val duration: Duration,
+    val playbackUris: PlatformSpecificUris,
+    val releaseYear: Int,
+    val genre: String,
     val episodeNumber: Int,
     val seasonNumber: Int,
     val showTitle: String,
     var nextEpisode: TvEpisodeEntity?,
 ) : VideoEntity {
-    override val type = EntityType.TV_EPISODE
+    override val type = EntityType.TvEpisode
 
-    fun toPlaybackEntity() = toPlaybackEntity(playbackPosition = null)
     override fun toPlaybackEntity(playbackPosition: Duration?) = PlaybackEntity(
         entityId = id,
         title = name,
+        duration = duration,
         releaseYear = releaseYear,
+        genre = genre,
+        playbackPosition = playbackPosition ?: 0.seconds
+    )
+}
+
+data class VideoClipEntity(
+    override val id: String,
+    override val name: String,
+    val images: List<Image>,
+    val duration: Duration,
+    val playbackUris: PlatformSpecificUris,
+    val releaseYear: Int,
+    val genre: String,
+    val creatorImage: String,
+    val creator: String,
+    var nextVideoClipEntity: VideoClipEntity?,
+) : VideoEntity {
+    override val type = EntityType.VideoClip
+
+    override fun toPlaybackEntity(playbackPosition: Duration?) = PlaybackEntity(
+        entityId = id,
+        title = name,
         duration = duration,
+        releaseYear = releaseYear,
+        genre = genre,
+        playbackPosition = playbackPosition ?: 0.seconds
+    )
+}
+
+data class TvShowEntity(
+    override val id: String,
+    override val name: String,
+    val images: List<Image>,
+    val description: String,
+    val platformSpecificUris: PlatformSpecificUris,
+    val releaseYear: Int,
+    val genre: String,
+    val seasonCount: Integer,
```

Comment:
Let's remove the season count for now. When we want to make use of seasons, we can create a new entity called `TvSeasonEntity` and then add a new field to the `TvShowEntity` called `val seasons: List<TvSeasonEntity>`. Then, we can create a getter which returns the `seasonCount`.

```kt
data class TvShowEntity(
  ...
) {
  val seasonCount = seasons.size()
}
```

This way, we will avoid having fragmented state and it will help us prevent out-of-sync issues.